main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -925,7 +925,7 @@ object Weeder2 {
       // Note: This visitor is used by both expression literals and pattern literals.
       expectAny(tree, List(TreeKind.Expr.Literal, TreeKind.Pattern.Literal))
       tree.children(0) match {
-        case Child.TokenChild(token) => token.kind match {
+        case token@Token(_, _, _, _, _, _ ,_ ,_) => token.kind match {
```

Comment:
But why are you binding it? Why not take out the kind directly as a variable and match on it :)